### ISSUE TITLE
jwt_optional throws an Exception on a refresh token

### ISSUE THREAD

**Issue by Guthers** [vimalloc/flask-jwt-extended#382]
Hello,

I ran into the issue of jwt_optional throwing an exception on refresh tokens.

I also saw #183. I would ask you to re-consider having no way to fix this issue other than a custom decorator.

However, if you do decide to not change the existing decorator, or add an additional one. Could you please make it super obvious in the docs that it will throw an exception.

Thanks.

Also here is my version of your all_jwt_optional from #183.

```python
def jwt_optional_with_refresh(fn):
    @wraps(fn)
    def wrapper(*args, **kwargs):
        try:
            verify_jwt_in_request()
        except (NoAuthorizationError, InvalidHeaderError):
            pass
        except WrongTokenError:
            try:
                verify_jwt_refresh_token_in_request()
            except(NoAuthorizationError, InvalidHeaderError):
                pass
        return fn(*args, **kwargs)
    return wrapper
```

**Comment by vimalloc** [comment]
This is already resolved in the 4.0.0 branch. In that upcoming release, `@jwt_required(refresh=True, optional=True)` is completely valid! :+1:

**Comment by Guthers** [comment]
Excellent, thanks :)
